client/aggregator.go

```diff
@@ -9,12 +9,18 @@ import (
 
 // newWatchAggregator maintains state of consumers calling `Watch` so that a
 // single `watch` request is made to the underlying client.
-func newWatchAggregator(c Client, l log.Logger) *watchAggregator {
-	return &watchAggregator{
+func newWatchAggregator(c Client, autoStart bool) *watchAggregator {
+	aggregator := &watchAggregator{
 		Client:      c,
-		log:         l,
+		autoStart:   autoStart,
+		log:         log.DefaultLogger,
 		subscribers: make([]subscriber, 0),
 	}
+	if autoStart {
+		ctx, cancel := context.WithCancel(context.Background())
+		go aggregator.distribute(aggregator.Client.Watch(ctx), cancel, true)
```

Comment:
Why not just call `aggregator.Watch(ctx)` here?

Reply:
because then an additional go routine would need to exist to keep the results channel from that watch call drained.